core/src/main/scala/com/pingcap/tispark/write/TiBatchWriteTable.scala

```diff
@@ -215,103 +215,73 @@ class TiBatchWriteTable(
     // check value not null
     checkValueNotNull(tiRowRdd)
 
+    val wrappedRowRdd: RDD[WrappedRow] = if (isCommonHandle || tiTableInfo.isPkHandle) {
+      val noDistinctWrappedRowRdd = tiRowRdd.map { row =>
+        WrappedRow(row, WriteUtil.extractHandle(row, tiTableInfo))
+      }
+      val distinctWrappedRowRdd = deduplicate(noDistinctWrappedRowRdd)
+      if (!options.deduplicate && noDistinctWrappedRowRdd.count() != distinctWrappedRowRdd
+          .count()) {
+        throw new TiBatchWriteException("duplicate unique key or primary key")
+      }
+      distinctWrappedRowRdd
+    } else {
+      val rowIDAllocator = getRowIDAllocator(count)
+      tiRowRdd.zipWithIndex.map { row =>
+        val index = row._2 + 1
+        val rowId = rowIDAllocator.getShardRowId(index)
+        WrappedRow(row._1, new IntHandle(rowId))
```

Comment:
I think we need to do deduplicate when uniqueIndices.nonEmpty here since you just judge `if (isCommonHandle || tiTableInfo.isPkHandle)`

Reply:
> I think we need to do deduplicate when uniqueIndices.nonEmpty here since you just judge `if (isCommonHandle || tiTableInfo.isPkHandle)`

changed